run/jobs/main.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"time"
+)
+
+type EnvVars struct {
+	// Job-defined env vars
+	taskNum    string
+	attemptNum string
+
+	// User-defined env vars
+	sleepMs  int64
+	failRate float64
+}
+
+func stringToInt(s string) int64 {
+	num, _ := strconv.ParseInt(s, 10, 64)
+	return num
+}
+
+func stringToFloat(s string) float64 {
+	num, _ := strconv.ParseFloat(s, 64)
+	return num
+}
+
+func main() {
```

Comment:
If this were a "real" app, I'd have main() just assemble the config values and log the task/attempt.

Collapsing the "core work" into main probably helps overall readability, and since this is a more "getting started" sample I'm in favor of the approach here.

Reply:
Ack.